File: Java/Data Structure/002. Insertion Sort.java
import java.util.Scanner;
public class Insertion {
    int i, j, n, temp;
    void InsertionSort(int arr[]){
        for(i=1; i<n; i++){
            temp = arr[i];
            j=i-1;
            while(j>=0 && arr[j]>temp){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = temp;
        }
    }
    void PrintArray(int arr[]){
        for(i=0; i<n; i++){
            System.out.print(arr[i]+"\t");
        }System.out.println();
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Insertion obj = new Insertion();
        System.out.println("Enter Array Size: ");
        obj.n=sc.nextInt();
        int arr[] = new int[obj.n];
        System.out.println("Enter "+obj.n+" Number:");
        for(int i=0; i<obj.n; i++){
            arr[i] = sc.nextInt();
        }
        System.out.println("Befor Sort: ");
        obj.PrintArray(arr);
        obj.InsertionSort(arr);
        System.out.println("After Insertion Sort: ");
        obj.PrintArray(arr);
    }
}

File: C++/Data Structure/002. Insertion Sort.cpp
#include<iostream>
using namespace std;
int i, j, n, temp;
void InsertionSort(int arr[]){
    for(i=1; i<n; i++){
        temp = arr[i];
        j=i-1;
        while(j>=0 && arr[j]>temp){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = temp;
    }
}
void PrintArray(int Arr[]){
    for(i=0; i<n; i++){
        cout<<Arr[i]<<"\t";
    }cout<<endl;
}
int main(){
    cout<<"Enter size of Array:\n";
    cin>>n;
    int arr[n];
    cout<<"Enter "<<n<<" Number:"<<endl;
    for(i=0; i<n; i++){
        cin>>arr[i];
    }
    cout<<"Before sort:\n";
    PrintArray(arr);
    InsertionSort(arr);
    cout<<"After Insertion sort:\n";
    PrintArray(arr);
}
